pandas/core/generic.py

```diff
@@ -4418,6 +4422,97 @@ def sort_values(
         """
         raise AbstractMethodError(self)
 
+    def sort_index(
+        self,
+        axis=0,
+        level=None,
+        ascending: bool_t = True,
+        inplace: bool_t = False,
+        kind: str = "quicksort",
+        na_position: str = "last",
+        sort_remaining: bool_t = True,
+        ignore_index: bool_t = False,
+        key: IndexKeyFunc = None,
+    ):
+
+        inplace = validate_bool_kwarg(inplace, "inplace")
+
+        is_dataframe = isinstance(self, ABCDataFrame)
+        if is_dataframe:
+            axis = self._get_axis_number(axis)
+            labels = self._get_axis(axis)
+            labels = ensure_key_mapped(labels, key, levels=level)
+            labels = labels._sort_levels_monotonic()
+            target = labels
+        else:
+            _ = self._get_axis_number(axis)
+            index = ensure_key_mapped(self.index, key, levels=level)
```

Comment:
I think you can shrink this down L4441-4451 to code that's the same for dataframe and series